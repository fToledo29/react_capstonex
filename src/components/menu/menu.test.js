import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'
import 'enzyme-to-json';
import { shallow } from 'enzyme';
import Menu from './menu';

describe('Menu', () => {

	const middlewares = [thunk];
	const mockStore = configureStore(middlewares);	
	let wrapper;
	let store;
	let props;

	beforeAll(() => {
		store = mockStore({
			data: {
				products: [],
				productsCopy: [],
				productToUpdate: {},
				productsToDelete: [],
				productViewMode: false,
			},
			userData: {},
			visitData: {},
			fieldsData: {},
		});

		props = {
			store: store,
			vistActions: {} ,
			actions: {} ,
			visitData: {} ,
			product: {} ,
			match: {params: {id: 1}, isExact: true, path: "", url: ""},
		};

		wrapper = shallow(<Menu {...props}/>).dive().dive();
	});

	it('Menu Renders correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('Home Menu Option', () => {
		expect(wrapper.find('.nav-link .login-label').text()).toContain(`Home`);
	});

	it('About Menu Option', () => {
		expect(wrapper.find('.nav-link.menu-about').text()).toContain(`About`);
	});

	it('Products Menu Option', () => {
		expect(wrapper.find('.nav-link.menu-products').text()).toContain(`Products`);
	});

	it('Top Viewed Products Menu Option', () => {
		expect(wrapper.find('.nav-link.menu-chart').text()).toContain(`Top Viewed Products`);
	});

	it('Login Menu Option', () => {
		expect(wrapper.find('.nav-buttons .login-label.login').text()).toContain(`Login`);
	});

	it('Register Menu Option', () => {
		expect(wrapper.find('.nav-buttons .login-label.register').text()).toContain(`Register`);
	});
});
