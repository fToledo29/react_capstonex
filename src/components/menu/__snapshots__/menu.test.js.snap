// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Menu Renders correctly 1`] = `
<div>
  <Navbar
    animation="false"
    bg="dark"
    collapseOnSelect={false}
    expand="lg"
    variant="dark"
  >
    <NavbarToggle
      animation="false"
      aria-controls="basic-navbar-nav"
      label="Toggle navigation"
    />
    <NavbarCollapse
      animation="false"
      id="basic-navbar-nav"
    >
      <Nav
        className="mr-auto"
        fill={false}
        justify={false}
      >
        <NavLink
          activeClassName="active"
          className="nav-link"
          exact={true}
          to="/"
        >
          <FontAwesomeIcon
            border={false}
            className="icons"
            fixedWidth={false}
            flip={null}
            icon="home"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size="1x"
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
          <span
            className="login-label"
          >
            Home
          </span>
        </NavLink>
        <NavLink
          activeClassName="active"
          className="nav-link menu-about"
          to="/about"
        >
          About
        </NavLink>
        <NavLink
          activeClassName="active"
          className="nav-link menu-products"
          to="/products"
        >
          Products
        </NavLink>
        <NavLink
          activeClassName="active"
          className="nav-link menu-chart"
          to="/chart"
        >
          Top Viewed Products
        </NavLink>
      </Nav>
      <Form
        className="nav-buttons"
        inline={true}
      >
        <Button
          active={false}
          disabled={false}
          variant="outline-info"
        >
          <NavLink
            activeClassName="active"
            to="/login"
          >
            <FontAwesomeIcon
              border={false}
              className="icons"
              fixedWidth={false}
              flip={null}
              icon="sign-in-alt"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size="1x"
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
            <span
              className="login-label login"
            >
              Login
            </span>
          </NavLink>
        </Button>
        <Button
          active={false}
          disabled={false}
          variant="outline-info"
        >
          <NavLink
            activeClassName="active"
            to="/register"
          >
            <FontAwesomeIcon
              border={false}
              className="icons"
              fixedWidth={false}
              flip={null}
              icon="user-plus"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size="1x"
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            />
            <span
              className="login-label register"
            >
              Register
            </span>
          </NavLink>
        </Button>
      </Form>
    </NavbarCollapse>
  </Navbar>
</div>
`;
